uniffi_macros/src/lib.rs

```diff
@@ -144,6 +144,90 @@ pub fn include_scaffolding(component_name: TokenStream) -> TokenStream {
     }.into()
 }
 
+struct FfiConverterRecordInput {
+    ident: syn::Ident,
+    fields: syn::Fields,
+}
+
+impl syn::parse::Parse for FfiConverterRecordInput {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let ident = input.parse()?;
+        let fields: syn::FieldsNamed = input.parse()?;
+        Ok(FfiConverterRecordInput {
+            ident,
+            fields: fields.into(),
+        })
+    }
+}
+
+/// Generate the FfiConverter implementation for a Record
+#[proc_macro]
+pub fn ffi_converter_record(input: TokenStream) -> TokenStream {
+    let input = syn::parse_macro_input!(input as FfiConverterRecordInput);
+    record::record_ffi_converter_impl(&input.fields, &input.ident).into()
+}
+
+struct FfiConverterEnumInput {
+    ident: syn::Ident,
+    variants: syn::punctuated::Punctuated<syn::Variant, syn::Token![,]>,
+}
+
+impl syn::parse::Parse for FfiConverterEnumInput {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let ident = input.parse()?;
+        let content;
+        syn::braced!(content in input);
+        let variants =
+            <syn::punctuated::Punctuated<syn::Variant, syn::Token![,]>>::parse_terminated(
```

Comment:
The exact type should be inferrable from the return expression.

```suggestion
            syn::punctuated::Punctuated::parse_terminated(
```